### PR TITLE
Fix: update when create a room with a question stack

### PR DESCRIPTION
**Pull request by LeviathanST**

### CODE REVIEW

File: src/rooms/rooms.service.ts [hunk 1]
```diff
@@ -112,10 +110,30 @@ export class RoomsService {
         message: 'Not found stack with this id or this stack is not active',
       });
     }
+    if (questionStack.status != QuestionStackStatus.ACTIVE) {
```
Comment (AnataAria) [inline src/rooms/rooms.service.ts]: !==

File: src/rooms/rooms.service.ts [hunk 2]
```diff
@@ -112,10 +110,30 @@ export class RoomsService {
...
+      });
+    }
+    if (info.type != questionStack.type) {
+      errs.push({
+        at: 'questionStackType',
+        message: "Question stack's and room's type are not match!",
+      });
+    }
+
     if (errs.length > 0) {
```
Comment (AnataAria) [inline src/rooms/rooms.service.ts]: u can use !errs.length

File: src/rooms/rooms.service.ts [hunk 3]
```diff
@@ -112,10 +110,30 @@ export class RoomsService {
...
       return [null, errs];
     }
     try {
+      await this.questionStackRepository
+        .createQueryBuilder()
+        .update(questionStack)
+        .set({ status: QuestionStackStatus.USED })
+        .where('id= :id', { id: questionStack.id })
+        .execute();
+
```
Comment (AnataAria) [inline src/rooms/rooms.service.ts]: Put this will make this query can't roll back when bug come

File: src/rooms/rooms.service.ts (hunk 2 above)
Comment (LeviathanST) [reply src/rooms/rooms.service.ts]: resolved

File: src/rooms/rooms.service.ts (hunk 3 above)
Comment (LeviathanST) [reply src/rooms/rooms.service.ts]: resolved

File: src/rooms/rooms.service.ts (hunk 1 above)
Comment (LeviathanST) [reply src/rooms/rooms.service.ts]: resolved

**Pull request closed without merging**